### PR TITLE
Update developer documentation: `yalc`, development workflow

### PR DESCRIPTION
**Pull request by lyzadanger**
Update development guide with additions about workflow; update section
on how to remove a yalc-linked package from an application project.

### CODE REVIEW

File: docs/developing.md [hunk 1]
```diff
@@ -1,58 +1,54 @@
...
 
-If you wish to revert back to the current installed version of `frontend-shared`, simply run:
+Remove `yalc` linking from an application and restore the project-defined
+published package version:
 
 ```shell
-# from the client dir
-$ yarn --check-files
+$ cd <application project directory>
+$ yalc remove @hypothesis/frontend-shared
```
Comment (lyzadanger) [inline docs/developing.md:53]: This is the most salient change. I'm not convinced `yarn --check-files` does what this intends; I am more certain that `yalc remove` _does_.

Comment (robertknight) [reply docs/developing.md:53]: It looks to me like `yalc remove` and `yarn --check-files` do different things. `yalc remove` removes the copy of the package in the project's `.yalc` dir, but does't appear to update the contents of `node_modules/`, which is what matters most. `yarn --check-files` ensures that the contents of `node_modules/` matches `yarn.lock` (ie. reset all files to published packages from npm) but doesn't touch the `.yalc` directory.

In other words, it doesn't look like `yalc remove` "restores the project-defined package version".

```sh
# Initial state - `@hypothesis/frontend-shared` installed from npm
$ ls -lh node_modules/@hypothesis/ | grep frontend-shared
drwxr-xr-x  8 robert  staff   256B 26 Jul 08:52 frontend-shared

# After running `yarn link`, `node_modules/hypothesis-shared` becomes a symlink
$ yalc link @hypothesis/frontend-shared
Package @hypothesis/frontend-shared@3.6.0+c970582d linked ==> /Users/robert/hypothesis/client/node_modules/@hypothesis/frontend-shared
$ ls -lh node_modules/@hypothesis/ | grep frontend-shared
lrwxr-xr-x  1 robert  staff    65B 26 Jul 08:52 frontend-shared -> /Users/robert/hypothesis/client/.yalc/@hypothesis/frontend-shared
$ ls -lh .yalc/@hypothesis/
total 0
drwxr-xr-x  9 robert  staff   288B 26 Jul 08:52 frontend-shared

# Running `yalc remove` removes the copy from `.yalic/`, but doesn't update the symlink in `node_modules/`
$ yalc remove @hypothesis/frontend-shared
Removing installation of @hypothesis/frontend-shared in /Users/robert/hypothesis/client
$ ls -lh node_modules/@hypothesis/ | grep frontend-shared
lrwxr-xr-x  1 robert  staff    65B 26 Jul 08:52 frontend-shared -> /Users/robert/hypothesis/client/.yalc/@hypothesis/frontend-shared
$ ls -lh .yalc/

# Running `yarn --check-files` does replace the symlink, but doesn't touch any files in `.yalc`.
$ yarn --check-files
yarn install v1.22.10
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...

✨  Done in 14.32s.
$ ls -lh node_modules/@hypothesis/ | grep frontend-shared
drwxr-xr-x  8 robert  staff   256B 26 Jul 08:53 frontend-shared
```

**Review by robertknight — CHANGES_REQUESTED** [review]
In my testing with the latest version of yalc (1.0.0-pre.53), `yalc remove` doesn't do what the README says. It removes the copy of the package from the consuming project's `.yalc` directory, but doesn't update `node_modules/@hypothesis/frontend-shared` to match the version specified in package.json/yarn.lock.

The other changes in this PR LGTM.

File: docs/developing.md (hunk 1 above)
Comment (lyzadanger) [reply docs/developing.md:53]: > It looks to me like yalc remove and yarn --check-files do different things. yalc remove removes the copy of the package in the project's .yalc dir, but does't appear to update the contents of node_modules/, which is what matters most. yarn --check-files ensures that the contents of node_modules/ matches yarn.lock (ie. reset all files to published packages from npm) but doesn't touch the .yalc directory.

> In other words, it doesn't look like yalc remove "restores the project-defined package version".

I wish `yalc`'s docs were a little clearer on these points (I know, I'm casting blame around a bit here).

Let's say the goal in a given local project repo (e.g. `lms`, `client`) is to make sure that project is using the package version for `@hypothesis/frontend-shared` as defined by `yarn.lock` (or `package.json`, if need be), and that all `.yalc` metadata is cleaned up. i.e. "Restore this project's dependencies, don't yalc-ify anything". 

To kind of regurgitate what I think you've written here: Sounds like `yarn --check-files` will "actually update the `node_modules` contents" to the package version in `yarn.lock`, but that leaves some `.yalc` floating around. `yalc remove` removes that `.yalc` metadata. Correct?

In which case, is it necessary, perhaps, to use _both_ commands to fully clean up?

Comment (robertknight) [reply docs/developing.md:53]: > In which case, is it necessary, perhaps, to use both commands to fully clean up?

Yes. However assuming that the developer only really cares about updating `node_module` to match yarn.lock, `yarn --check-files` is the only command of the two that is essential.

**Review by robertknight — APPROVED** [review]
LGTM

**Pull request merged**